README.md

```diff
@@ -50,6 +50,28 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described [here](https://sanctuary.js.org/#types). In summary:
+
+1. New types can be created via type constructors. Type constructors take one or more type arguments. For example, the type "array of strings" is written as `Array String`, and the type constructor `Array` can be defined as `Array :: a -> a`.
```

Comment:
> A lot hinges on properly understanding `Type`.

I think introducing a `Type` type actually makes things less clear. Another option is to provide a few examples to encourage an intuitive understanding rather than a theoretical one:

>   - `Array String` is the type comprising every array of strings. Examples:
>
>       - `[]`
>       - `['foo', 'bar', 'baz']`
>
>   - `Array (Array String)` is the type comprising every array of array of strings. Examples:
>
>       - `[]`
>       - `[[], [], [], []]`
>       - `[[], ['foo'], ['bar', 'baz']]`